src/renderers/dom/shared/DOMProperty.js

```diff
@@ -208,22 +219,51 @@ var DOMProperty = {
   getPossibleStandardName: __DEV__ ? {autofocus: 'autoFocus'} : null,
 
   /**
-   * All of the isCustomAttribute() functions that have been injected.
-   */
-  _isCustomAttributeFunctions: [],
-
-  /**
-   * Checks whether a property name is a custom attribute.
+   * Checks whether a property name is a writeable attribute.
    * @method
    */
-  isCustomAttribute: function(attributeName) {
-    for (var i = 0; i < DOMProperty._isCustomAttributeFunctions.length; i++) {
-      var isCustomAttributeFn = DOMProperty._isCustomAttributeFunctions[i];
-      if (isCustomAttributeFn(attributeName)) {
+  shouldSetAttribute: function(name, value) {
+    if (
+      DOMProperty.isReservedProp(name) ||
+      DOMProperty.aliases.hasOwnProperty(name)
+    ) {
+      return false;
+    }
+
+    if (DOMProperty.properties.hasOwnProperty(name)) {
+      return true;
+    }
+
+    if (value === null) {
+      return true;
+    }
+
+    if (name.toLowerCase() !== name) {
+      return false;
+    }
```

Comment:
I left a bit of a text book at the bottom of the PR 😰. The only alternative I can think of is to let case insensitive attributes fall through and  warn about them. If an attribute has an alias, like `class` and `className`, we should prevent cases like `classname` from falling through. 

In the long term, this would allow something like `allowtransparency` in the DOM just like `allowTransparency`, but React would warn about it. Otherwise, we're stuck with the whitelist forever because we need to check for proper casing.